cosmetics-web/spec/controllers/component_build_controller_spec.rb

```
@@ -74,6 +75,48 @@
       expect(assigns(:component).cmrs).to have(5).items
       expect(assigns(:component).cmrs).to all(have_attributes(name: be_nil))
     end
+
+    describe "upload product ingredients page" do
+      before { get(:show, params: params.merge(id: :upload_formulation)) }
+
+      render_views
+
+      describe "page title" do
+        context "with a component with predefined formulation" do
+          let(:component_type) { "predefined" }
+
+          it "sets the page title to poisonous ingredients" do
+            expect(response.body).to match(/<title>Upload list of poisonous ingredients .+<\/title>/)
```

These will become brittle when we have to change the page titles 🤔

---

Yes, I did check the locale file but it appears we're not using it at the moment, all the text being hard coded in the view. I think we should look at that separately.

---

I personally would expect tests to fail if the page titles change. It does make them more brittle, but the tests will fail fast and be easy to update. Plus it makes them more readable, and allows you to refactor or reorganise the locale file without breaking the tests.